src/MSONUtility.h

```diff
@@ -121,26 +121,28 @@ namespace mson {
      *
      * \param subject String which represents the type name
      * \param typeName MSON Type Name
+     * \param isBaseType If false, will be parsed as a symbol
      */
     inline void parseTypeName(const std::string& subject,
-                              TypeName& typeName) {
+                              TypeName& typeName,
+                              bool isBaseType = true) {
```

Comment:
Better solution for readability is change condition:
```
if(!(isBaseType) {
  typeName.symbol = parseSymbol(subject)
} else {
  if (subject == "boolean") {
    typeName.base = BooleanTypeName;  
  } else if(...) {
  }   
}
```

And idealy move `else` into own method